connector/otlpjsonconnector/logs.go

```diff
@@ -50,16 +50,26 @@ func (c *connectorLogs) ConsumeLogs(ctx context.Context, pl plog.Logs) error {
 			for k := 0; k < logRecord.LogRecords().Len(); k++ {
 				lRecord := logRecord.LogRecords().At(k)
 				token := lRecord.Body()
-				var l plog.Logs
-				l, err := logsUnmarshaler.UnmarshalLogs([]byte(token.AsString()))
-				if err != nil {
-					c.logger.Error("could not extract logs from otlp json", zap.Error(err))
+
+				// Check if the "resourceLogs" key exists in the JSON data
+				if logRegex.MatchString(token.AsString()) {
+					var l plog.Logs
+					l, err := logsUnmarshaler.UnmarshalLogs([]byte(token.AsString()))
+					if err != nil {
+						c.logger.Error("could not extract logs from otlp json", zap.Error(err))
+						continue
+					}
+					err = c.logsConsumer.ConsumeLogs(ctx, l)
+					if err != nil {
+						c.logger.Error("could not consume logs from otlp json", zap.Error(err))
+					}
+				} else if metricRegex.MatchString(token.AsString()) || traceRegex.MatchString(token.AsString()) {
```

Comment:
actually surprisingly, it did not change much. I have the implementation and benchmarks here. You can take a look 

https://github.com/khushijain21/opentelemetry-collector-contrib/pull/2

I used `strings.HasPrefix` to get a match. Is there any other way we can extract the value? Appreciate your help

Reply:
@djaglowski any update?

Reply:
Is the plan to include the [benchmark](https://github.com/khushijain21/opentelemetry-collector-contrib/pull/2/files#diff-3b620795376706b622e07068d80b129a031f7b46c14709c26f9cdf26e536ce5dR182) as part of this PR? I think it would make sense since we have it.

Reply:
@ChrsMark can you take a look now

Reply:
Thank's! @djaglowski is this what you had in mind for the benchmark?

Reply:
Thanks for adding the benchmark. I played with it a bit and noticed that the cost only changes much when the payloads don't match any regex, which indicates to me that the cost of unmarshaling far exceed the cost of evaluating the regexes.

Reply:
But we can still "optimize" a bit here and only call the `token.AsString()` only once, right? (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35827/files#r1832545857).